tests/test-awful-rules.lua

```diff
@@ -0,0 +1,290 @@
+local awful = require("awful")
+local beautiful = require("beautiful")
+local test_client = require("_client")
+local unpack = unpack or table.unpack -- luacheck: globals unpack (compatibility with Lua 5.1)
+
+local callback_called = false
+
+-- Magic table to store tests
+local tests = {}
+
+local tb_height = awful.util.round(beautiful.get_font_height() * 1.5)
+-- local border_width = beautiful.border_width
+
+local function test_rule(rule)
+    rule.rule = rule.rule or {}
+
+    -- Create a random class. The number need to be large as "rule1" and "rule10" would collide
+    -- The math.ceil is necessary to remove the floating point, this create an invalid dbus name
+    rule.rule.class = rule.rule.class or "rule"..#tests..math.ceil(math.random()*100000000)
```

Comment:
Ok, then use `string.format("%010d", counter)`, that will guarantee uniqueness for quite a while.